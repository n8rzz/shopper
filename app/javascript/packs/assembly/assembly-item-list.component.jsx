import React from 'react';
import PropTypes from 'prop-types';
import { AssemblyItemContainer } from './assembly-item.container';

export function AssemblyItemList(props) {
    return (
        <ul className={'vlist mix-vlist_isClickable js-accordionList'}>
            {props.assemblies.map((assembly, index) => (
                <AssemblyItemContainer
                    key={`assemblyItemContainer-${index}`}
                    assembly={assembly}
                    assemblyItemsGroupedByAssemblyId={props.assemblyItemsGroupedByAssemblyId}
                    csrf={props.csrf}
                    departmentIdMap={props.departmentIdMap}
                    itemIdMap={props.itemIdMap}
                />
            ))}
        </ul>
    );
}

AssemblyItemList.propTypes = {
    assemblies: PropTypes.array.isRequired,
    assemblyItemsGroupedByAssemblyId: PropTypes.object.isRequired,
    csrf: PropTypes.string.isRequired,
    departmentIdMap: PropTypes.object.isRequired,
    itemIdMap: PropTypes.object.isRequired,
};
