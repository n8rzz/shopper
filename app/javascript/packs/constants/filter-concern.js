export const FILTER_CONCERN = {
    ALPHA: 'ALPHA',
    DEPARTMENT: 'DEPARTMENT',
    FILTER: 'FILTER',
    ITEM: 'ITEM',
    MEAL: 'MEAL',
};

export const FILTER_LABEL = {
    [FILTER_CONCERN.ALPHA]: 'Alpha',
    [FILTER_CONCERN.DEPARTMENT]: 'Department',
    [FILTER_CONCERN.FILTER]: 'Filter',
    [FILTER_CONCERN.ITEM]: 'Item',
    [FILTER_CONCERN.MEAL]: 'Meal',
};
