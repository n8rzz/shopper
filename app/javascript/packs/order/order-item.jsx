import React from 'react';
import PropTypes from 'prop-types'
import classnames from 'classnames';
import {noop} from 'lodash/noop';

export class OrderItem extends React.Component {
    constructor(props) {
        super(props);
    }

    _buildOrderItemClassnames() {
        const orderItemIdClassname = `js-orderItem-${this.props.orderItemId}`;

        return classnames({
            'orderItem': true,
            [orderItemIdClassname]: true,
            'mix-orderItem_isPicked': this.props.isPicked,
        });
    }

    _buildCheckboxJsx() {
        return (
            <div className="orderItem-action">
                <form>
                    <input
                        type="checkbox"
                        className="input-checkbox js-orderItem-action-value u-isVisuallyHidden"
                        name={this.props.orderItemId}
                        checked={this.props.isPicked}
                        onChange={() => noop}
                    />
                    <label
                        className="input-label_checkbox js-orderItem-action"
                        htmlFor={this.props.orderItemId}
                        onClick={this.props.onClickCheckboxHandler}
                    >
                        {this.props.isPicked}
                    </label>
                </form>
            </div>
        );
    }

    _buildItemTitleJsx() {
        return (
            <div className="orderItem-hd-content">
                <h3>
                    <a
                        href={this.props.editItemUrl}
                        className="orderItem-hd-link"
                    >
                        {this.props.itemName}
                    </a>
                </h3>
            </div>
        );
    }

    _buildQtyJsx() {
        if (this.props.qty < 2) {
            return null;
        }

        return (
            <div className="orderItem-hd-supplement">
                <h3>
                    <span className="txt-light">Qty:</span> {this.props.qty}
                </h3>
            </div>
        );
    }

    _buildRemoveItemButtonJsx() {
        return (
            <button
                type="submit"
                className="icon-btn js-orderItem-remove"
                onClick={this.props.onClickRemoveItemHandler}
            >
                <span className="icon icon-trashcan"></span>
            </button>
        );
    }

    _buildAssemblyTitleJsx() {
        if (this.props.assemblyName === '') {
            return null;
        }

        return (
            <div className="orderItem-bd">
                {this.props.assemblyName}
            </div>
        );
    }

    render() {
        return (
            <li className={this._buildOrderItemClassnames()}>
                {this._buildCheckboxJsx()}

                <div className="orderItem-content">
                    <div className="orderItem-hd">
                        {this._buildItemTitleJsx()}
                        {this._buildQtyJsx()}
                        {this._buildRemoveItemButtonJsx()}
                    </div>
                    {this._buildAssemblyTitleJsx()}
                    <div className="orderItem-ft">
                        {this.props.departmentName}
                    </div>
                </div>
            </li>
        );
    }
}

OrderItem.propTypes = {
    assemblyName: PropTypes.string.isRequired,
    departmentName: PropTypes.string.isRequired,
    editItemUrl: PropTypes.string.isRequired,
    isPicked: PropTypes.bool.isRequired,
    itemName: PropTypes.string.isRequired,
    orderItemId: PropTypes.number.isRequired,
    qty: PropTypes.number.isRequired,
    onClickCheckboxHandler: PropTypes.func.isRequired,
    onClickRemoveItemHandler: PropTypes.func.isRequired,
};
