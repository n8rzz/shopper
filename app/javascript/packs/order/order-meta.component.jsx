import React from 'react';
import PropTypes from 'prop-types';
import format from 'date-fns/format';

export function OrderMeta(props) {
    const formattedShoppingDate = format(props.shoppingDate, 'MMM DD, YYYY');

    return (
        <div className={'orderMeta'}>
            <div>
                Shopping Date: { formattedShoppingDate }
            </div>
            <div>
                Status: { props.status }
            </div>
            <div>
                Location: { props.locationName }
            </div>
        </div>
    );
}

OrderMeta.propTypes = {
    shoppingDate: PropTypes.string.isRequired,
    status: PropTypes.string.isRequired,
    locationName: PropTypes.string,
};

OrderMeta.defaultProps = {
    locationName: '',
};
