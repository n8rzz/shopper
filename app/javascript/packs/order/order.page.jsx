import React from 'react';
import PropTypes from 'prop-types';
import _keyBy from 'lodash/keyBy';
import { FilterBar } from '../components/filter-bar/filter-bar.component';
import { OrderItemList } from './order-item-list.component';
import { OrderConcernGroup } from './order-concern-group.component';
import {
    FILTER_CONCERN,
    FILTER_LABEL,
} from '../constants/filter-concern';

const ORDER_LIST_RENDERER = {
    [FILTER_CONCERN.ITEM]: OrderItemList,
    [FILTER_CONCERN.DEPARTMENT]: OrderConcernGroup,
    [FILTER_CONCERN.MEAL]: OrderConcernGroup,
};

export const FILTER_BAR_ITEMS = [
    {
        text: FILTER_LABEL.ITEM,
        url: '',
    },
    {
        text: FILTER_LABEL.DEPARTMENT,
        url: '',
    },
    {
        text: FILTER_LABEL.MEAL,
        url: '',
    },
];

export class OrderPage extends React.Component {
    constructor(props) {
        super(props);

        this._assemblyMap = _keyBy(props.assemblies, 'id');
        this._departmentMap = _keyBy(props.departments, 'id');
        this._itemMap = _keyBy(props.items, 'id');
        this._onChangeFilterHandler = this._onChangeFilter.bind(this);
        this.state = {
            concern: FILTER_CONCERN.ITEM,
        };
    }

    _onChangeFilter(event) {
        event.preventDefault();

        const filterName = event.currentTarget.text.toUpperCase();

        this.setState({ concern: FILTER_CONCERN[filterName] });
    }

    render() {
        const OrderItemRenderer = ORDER_LIST_RENDERER[this.state.concern];

        return (
            <React.Fragment>
                <FilterBar
                    initialSelection={'Item'}
                    items={FILTER_BAR_ITEMS}
                    onChangeFilterHandler={this._onChangeFilterHandler}
                />
                <div className={'js-orderItem'}>
                    <OrderItemRenderer
                        assemblies={this.props.assemblies}
                        assemblyMap={this._assemblyMap}
                        concern={this.state.concern}
                        csrf={this.props.csrf}
                        departmentMap={this._departmentMap}
                        departments={this.props.departments}
                        itemMap={this._itemMap}
                        orderItems={this.props.orderItems}
                    />
                </div>
            </React.Fragment>
        );
    }
}

OrderPage.propTypes = {
    assemblies: PropTypes.array.isRequired,
    csrf: PropTypes.string.isRequired,
    departments: PropTypes.array.isRequired,
    items: PropTypes.array.isRequired,
    orderItems: PropTypes.array.isRequired,
};
