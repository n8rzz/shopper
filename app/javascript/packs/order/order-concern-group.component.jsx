import React from 'react';
import PropTypes from 'prop-types';
import _get from 'lodash/get';
import _groupBy from 'lodash/groupBy';
import _sortBy from 'lodash/sortBy';
import _uniqBy from 'lodash/uniqBy';
import { OrderItem } from './order-item.component';
import { FILTER_CONCERN } from '../constants/filter-concern';

export class OrderConcernGroup extends React.Component {
    constructor(props) {
        super(props);

        this._sortedAssemblyNameList = _sortBy(props.assemblies, 'name');
        this._sortedDepartmentNameList = this._buildSortedDepartmentList();
        this._groupedByAssemblyOrderItemList = _groupBy(props.orderItems, 'assembly_id');
        this._groupedByDepartmentOrderItemList = _groupBy(props.orderItems, 'department_id');
    }

    componentWillUpdate(nextProps) {
        this._groupedByAssemblyOrderItemList = _groupBy(nextProps.orderItems, 'assembly_id');
        this._groupedByDepartmentOrderItemList = _groupBy(nextProps.orderItems, 'department_id');
    }

    _buildSortedDepartmentList() {
        if (this.props.locationDepartments.length > 0) {
            return this._buildSortedDepartmentListWithSortOrder();
        }

        return this._buildSortedByNameDepartmentList();
    }

    _buildSortedDepartmentListWithSortOrder() {
        const departmentListBySortOrder = this.props.locationDepartments.map(
            (locationDepartment) => this.props.departmentMap[locationDepartment.department_id],
        );

        return _uniqBy([
            ...departmentListBySortOrder,
            ...this._buildSortedByNameDepartmentList(),
        ], 'id');
    }

    _buildSortedByNameDepartmentList() {
        return _sortBy(this.props.departments, 'name');
    }

    _buildOrderItemListForConcernGroupJsx(orderItems) {
        const orderItemListJsx = orderItems.map((orderItem, index) => {
            const assembly = this.props.assemblyMap[orderItem.assembly_id];
            const assemblyName = _get(assembly, 'name', '');
            const departmentName = this.props.departmentMap[orderItem.department_id].name;
            const itemName = this.props.itemMap[orderItem.item_id].name;
            const editItemUrl = `/item/${orderItem.item_id}/edit`;

            return (
                <OrderItem
                    assemblyName={assemblyName}
                    csrf={this.props.csrf}
                    departmentName={departmentName}
                    editItemUrl={editItemUrl}
                    isPicked={orderItem.picked}
                    itemName={itemName}
                    key={`orderItem-${index}`}
                    orderItemId={orderItem.id}
                    onClickIsPickedHandler={this.props.onClickIsPickedHandler}
                    onClickRemoveItemHandler={this.props.onClickRemoveItemHandler}
                    qty={orderItem.qty}
                />
            );
        });

        return (
            <ul className={'vlist'}>
                {orderItemListJsx}
            </ul>
        );
    }

    _buildConcernGroupJsx(name, index, itemList = []) {
        if (itemList.length === 0) {
            return null;
        }

        return (
            <div
                key={`orderItem-concern-${name}-${index}`}
                className={'concernGroup'}
            >
                <div className={'concernGroup-hd'}>
                    <h2>{name}</h2>
                </div>
                <div className={'concernGroup-bd'}>
                    {this._buildOrderItemListForConcernGroupJsx(itemList)}
                </div>
            </div>
        );
    }

    _buildOrderItemListForDepartmentConcernGroupJsx() {
        return this._sortedDepartmentNameList.reduce((sum, department, index) => {
            const itemsForDepartmentId = this._groupedByDepartmentOrderItemList[department.id];

            if (typeof itemsForDepartmentId === 'undefined') {
                return sum;
            }

            const orderItemConcernGroupJsx = this._buildConcernGroupJsx(department.name, index, itemsForDepartmentId);

            sum.push(orderItemConcernGroupJsx);

            return sum;
        }, []);
    }

    _buildOrderItemListForAssemblyConcernGroupJsx() {
        const assemblyItemsJsx = this._sortedAssemblyNameList.reduce((sum, assembly, index) => {
            const itemsForAssemblyId = this._groupedByAssemblyOrderItemList[assembly.id];

            if (typeof itemsForAssemblyId === 'undefined') {
                return sum;
            }

            const orderItemConcernGroupJsx = this._buildConcernGroupJsx(assembly.name, index, itemsForAssemblyId);

            sum.push(orderItemConcernGroupJsx);

            return sum;
        }, []);

        const nextIndexForNonGroupedItems = assemblyItemsJsx.length + 1;
        const itemsWithoutAssemblyJsx = this._buildConcernGroupJsx(
            'Not Grouped',
            nextIndexForNonGroupedItems,
            this._groupedByAssemblyOrderItemList.null,
        );

        assemblyItemsJsx.push(itemsWithoutAssemblyJsx);

        return assemblyItemsJsx;
    }

    render() {
        // TODO: rename this var
        let jsx = null;

        if (this.props.concern === FILTER_CONCERN.DEPARTMENT) {
            jsx = this._buildOrderItemListForDepartmentConcernGroupJsx();
        }

        if (this.props.concern === FILTER_CONCERN.MEAL) {
            jsx = this._buildOrderItemListForAssemblyConcernGroupJsx();
        }

        return jsx;
    }
}

OrderConcernGroup.propTypes = {
    assemblies: PropTypes.array.isRequired,
    assemblyMap: PropTypes.object.isRequired,
    concern: PropTypes.string.isRequired,
    csrf: PropTypes.string.isRequired,
    departmentMap: PropTypes.object.isRequired,
    departments: PropTypes.array.isRequired,
    itemMap: PropTypes.object.isRequired,
    locationDepartments: PropTypes.array.isRequired,
    orderItems: PropTypes.array.isRequired,
    onClickIsPickedHandler: PropTypes.func.isRequired,
    onClickRemoveItemHandler: PropTypes.func.isRequired,
};
