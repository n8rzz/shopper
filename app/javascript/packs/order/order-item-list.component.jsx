import React from 'react';
import PropTypes from 'prop-types';
import _get from 'lodash/get';
import { OrderItemContainer } from './order-item.container';

export function OrderItemList(props) {
    return (
        <ul className={'vlist'}>
            {props.orderItems.map((orderItem, index) => {
                const assembly = props.assemblyMap[orderItem.assembly_id];
                const assemblyName = _get(assembly, 'name', '');
                const departmentName = props.departmentMap[orderItem.department_id].name;
                const itemName = props.itemMap[orderItem.item_id].name;
                const editItemUrl = `/item/${orderItem.item_id}/edit`;

                return (
                    <OrderItemContainer
                        key={`orderItemContainer-${index}`}
                        csrf={props.csrf}
                        orderItemId={orderItem.id}
                        qty={orderItem.qty}
                        itemName={itemName}
                        editItemUrl={editItemUrl}
                        departmentName={departmentName}
                        assemblyName={assemblyName}
                        isPicked={orderItem.picked}
                    />
                );
            })}
        </ul>
    );
}

OrderItemList.propTypes = {
    assemblyMap: PropTypes.object.isRequired,
    csrf: PropTypes.string.isRequired,
    departmentMap: PropTypes.object.isRequired,
    itemMap: PropTypes.object.isRequired,
    orderItems: PropTypes.array.isRequired,
};
