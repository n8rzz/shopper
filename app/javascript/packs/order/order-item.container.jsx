import React from 'react';
import PropTypes from 'prop-types';
import ApiService from '../service/api.service';
import { OrderItem } from './order-item';

export class OrderItemContainer extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            isPicked: props.isPicked,
        };
    }

    componentWillMount() {
        this._onClickPickedCheckboxHandler = this._onClickPickedCheckbox.bind(this);
        this._onClickRemoveItemHandler = this._onClickRemoveItem.bind(this);
        this._onUpdatePickedSuccessHandler = this._onUpdatePickedSuccess.bind(this);
        this._onRemoveItemSuccessHandler = this._onRemoveItemSuccess.bind(this);
    }

    componentWillUnmount() {
        this._onClickPickedCheckboxHandler = null;
        this._onClickRemoveItemHandler = null;
        this._onUpdatePickedSuccessHandler = null;
        this._onRemoveItemSuccessHandler = null;
    }

    _onClickPickedCheckbox() {
        const orderItemUrl = `/order_items/${this.props.orderItemId}.json`;
        const itemUpdateToSend = {
            picked: !this.state.isPicked,
        };

        ApiService.patch(orderItemUrl, itemUpdateToSend, this.props.csrf)
            .then((response) => {
                if (response.status !== 200) {
                    console.error(`Received an unexpected status code from ${this.props.editItemUrl}. See response: ${response}`);

                    return;
                }

                this._onUpdatePickedSuccessHandler(response.data);
            })
            .catch((error) => { throw error; });
    }

    _onClickRemoveItem() {
        const orderItemUrl = `/order_items/${this.props.orderItemId}.json`;

        ApiService.delete(orderItemUrl, this.props.csrf)
            .then((response) => {
                if (response.status !== 204) {
                    console.error(`Received an unexpected status code from ${orderItemUrl}. See response: ${response}`);

                    return;
                }

                this._onRemoveItemSuccessHandler(this.props.orderItemId);
            })
            .catch((error) => { throw error; });
    }

    _onUpdatePickedSuccess() {
        this.setState((prevState) => ({ isPicked: !prevState.isPicked }));
    }

    _onRemoveItemSuccess(orderItemId) {
        console.log('+++ _onRemoveItemSuccess', orderItemId);
        console.log('--- this should bubble up and remove this item from the list ---');
    }

    render() {
        return (
            <OrderItem
                assemblyName={this.props.assemblyName}
                departmentName={this.props.departmentName}
                editItemUrl={this.props.editItemUrl}
                isPicked={this.state.isPicked}
                itemName={this.props.itemName}
                orderItemId={this.props.orderItemId}
                qty={this.props.qty}
                onClickCheckboxHandler={this._onClickPickedCheckboxHandler}
                onClickRemoveItemHandler={this._onClickRemoveItemHandler}
            />
        );
    }
}

OrderItemContainer.propTypes = {
    assemblyName: PropTypes.string.isRequired,
    csrf: PropTypes.string.isRequired,
    departmentName: PropTypes.string.isRequired,
    editItemUrl: PropTypes.string.isRequired,
    isPicked: PropTypes.bool.isRequired,
    itemName: PropTypes.string.isRequired,
    orderItemId: PropTypes.number.isRequired,
    qty: PropTypes.number.isRequired,
};
