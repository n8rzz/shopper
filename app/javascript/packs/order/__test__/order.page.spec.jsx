import React from 'react';
import { mount, shallow } from 'enzyme';
import { OrderPage } from '../order.page';
import { FILTER_CONCERN } from '../../constants/filter-concern';
import { assemblyListMock } from '../../__mocks__/assembly.mock';
import { csrfMock } from '../../__mocks__/csrf.mock';
import { departmentListMock } from '../../__mocks__/department.mock';
import { orderItemListMock } from '../../__mocks__/order-item.mock';
import { itemListMock } from '../../__mocks__/item.mock';

describe('OrderPage', () => {
    describe('when passed more than 0 `orderItems`', () => {
        test('renders correctly with valid props', () => {
            const component = shallow(<OrderPage
                assemblies={assemblyListMock}
                csrf={csrfMock}
                departments={departmentListMock}
                items={itemListMock}
                orderItems={orderItemListMock}
            />);

            expect(component).toMatchSnapshot();
        });
    });

    describe('when passed 0 `orderItems`', () => {
        test('renders correctly with valid props', () => {
            const component = mount(<OrderPage
                assemblies={assemblyListMock}
                csrf={csrfMock}
                departments={departmentListMock}
                items={itemListMock}
                orderItems={[]}
            />);
            component.setState({ concern: FILTER_CONCERN.MEAL });

            expect(component).toMatchSnapshot();
        });
    });
});
