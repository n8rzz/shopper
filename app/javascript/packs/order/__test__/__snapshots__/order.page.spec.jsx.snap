// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderPage when passed 0 \`orderItems\` renders correctly with valid props 1`] = `
<OrderPage
  assemblies={
    Array [
      Object {
        "created_at": "2019-07-29T22:49:47.431Z",
        "id": 1,
        "name": "Souvlaki",
        "updated_at": "2019-07-29T22:49:47.431Z",
      },
      Object {
        "created_at": "2019-07-29T22:49:47.519Z",
        "id": 2,
        "name": "Chilli con Carne",
        "updated_at": "2019-07-29T22:49:47.519Z",
      },
    ]
  }
  csrf="3yIq75tyV791TFkzSZo/S4b+TVa/N5J3dI14PzMUPviGViEGeDSnGnvNix+MaB0RB0XuV7bpX8I3SmIEAr4/zQ=="
  departments={
    Array [
      Object {
        "created_at": "2019-07-21T16:03:32.883Z",
        "id": 3,
        "name": "Bakery",
        "updated_at": "2019-07-21T16:03:32.883Z",
      },
      Object {
        "created_at": "2019-07-21T16:03:32.883Z",
        "id": 5,
        "name": "Meat",
        "updated_at": "2019-07-21T16:03:32.883Z",
      },
    ]
  }
  items={
    Array [
      Object {
        "created_at": "2019-07-21T16:03:33.162Z",
        "department_id": 5,
        "id": 15,
        "name": "Agave Syrup",
        "updated_at": "2019-07-21T16:03:33.162Z",
      },
      Object {
        "created_at": "2019-07-21T16:03:33.162Z",
        "department_id": 3,
        "id": 13,
        "name": "Agave Syrup",
        "updated_at": "2019-07-21T16:03:33.162Z",
      },
    ]
  }
  orderItems={Array []}
>
  <FilterBar
    initialSelection="Item"
    items={
      Array [
        Object {
          "text": "Item",
          "url": "",
        },
        Object {
          "text": "Department",
          "url": "",
        },
        Object {
          "text": "Meal",
          "url": "",
        },
      ]
    }
    onChangeFilterHandler={[Function]}
  >
    <ul
      className="filterBar"
    >
      <FilterBarItem
        isActive={true}
        key="filterBar-item-0"
        onChangeFilterHandler={[Function]}
        text="Item"
        url=""
      >
        <li>
          <a
            className="filterBar-link mix-filterBar-link_isActive"
            href=""
            onClick={[Function]}
          >
            Item
          </a>
        </li>
      </FilterBarItem>
      <FilterBarItem
        isActive={false}
        key="filterBar-item-1"
        onChangeFilterHandler={[Function]}
        text="Department"
        url=""
      >
        <li>
          <a
            className="filterBar-link"
            href=""
            onClick={[Function]}
          >
            Department
          </a>
        </li>
      </FilterBarItem>
      <FilterBarItem
        isActive={false}
        key="filterBar-item-2"
        onChangeFilterHandler={[Function]}
        text="Meal"
        url=""
      >
        <li>
          <a
            className="filterBar-link"
            href=""
            onClick={[Function]}
          >
            Meal
          </a>
        </li>
      </FilterBarItem>
    </ul>
  </FilterBar>
  <div
    className="js-orderItem"
  >
    <OrderConcernGroup
      assemblies={
        Array [
          Object {
            "created_at": "2019-07-29T22:49:47.431Z",
            "id": 1,
            "name": "Souvlaki",
            "updated_at": "2019-07-29T22:49:47.431Z",
          },
          Object {
            "created_at": "2019-07-29T22:49:47.519Z",
            "id": 2,
            "name": "Chilli con Carne",
            "updated_at": "2019-07-29T22:49:47.519Z",
          },
        ]
      }
      assemblyMap={
        Object {
          "1": Object {
            "created_at": "2019-07-29T22:49:47.431Z",
            "id": 1,
            "name": "Souvlaki",
            "updated_at": "2019-07-29T22:49:47.431Z",
          },
          "2": Object {
            "created_at": "2019-07-29T22:49:47.519Z",
            "id": 2,
            "name": "Chilli con Carne",
            "updated_at": "2019-07-29T22:49:47.519Z",
          },
        }
      }
      concern="MEAL"
      csrf="3yIq75tyV791TFkzSZo/S4b+TVa/N5J3dI14PzMUPviGViEGeDSnGnvNix+MaB0RB0XuV7bpX8I3SmIEAr4/zQ=="
      departmentMap={
        Object {
          "3": Object {
            "created_at": "2019-07-21T16:03:32.883Z",
            "id": 3,
            "name": "Bakery",
            "updated_at": "2019-07-21T16:03:32.883Z",
          },
          "5": Object {
            "created_at": "2019-07-21T16:03:32.883Z",
            "id": 5,
            "name": "Meat",
            "updated_at": "2019-07-21T16:03:32.883Z",
          },
        }
      }
      departments={
        Array [
          Object {
            "created_at": "2019-07-21T16:03:32.883Z",
            "id": 3,
            "name": "Bakery",
            "updated_at": "2019-07-21T16:03:32.883Z",
          },
          Object {
            "created_at": "2019-07-21T16:03:32.883Z",
            "id": 5,
            "name": "Meat",
            "updated_at": "2019-07-21T16:03:32.883Z",
          },
        ]
      }
      itemMap={
        Object {
          "13": Object {
            "created_at": "2019-07-21T16:03:33.162Z",
            "department_id": 3,
            "id": 13,
            "name": "Agave Syrup",
            "updated_at": "2019-07-21T16:03:33.162Z",
          },
          "15": Object {
            "created_at": "2019-07-21T16:03:33.162Z",
            "department_id": 5,
            "id": 15,
            "name": "Agave Syrup",
            "updated_at": "2019-07-21T16:03:33.162Z",
          },
        }
      }
      orderItems={Array []}
    />
  </div>
</OrderPage>
`;

exports[`OrderPage when passed more than 0 \`orderItems\` renders correctly with valid props 1`] = `
<Fragment>
  <FilterBar
    initialSelection="Item"
    items={
      Array [
        Object {
          "text": "Item",
          "url": "",
        },
        Object {
          "text": "Department",
          "url": "",
        },
        Object {
          "text": "Meal",
          "url": "",
        },
      ]
    }
    onChangeFilterHandler={[Function]}
  />
  <div
    className="js-orderItem"
  >
    <OrderItemList
      assemblies={
        Array [
          Object {
            "created_at": "2019-07-29T22:49:47.431Z",
            "id": 1,
            "name": "Souvlaki",
            "updated_at": "2019-07-29T22:49:47.431Z",
          },
          Object {
            "created_at": "2019-07-29T22:49:47.519Z",
            "id": 2,
            "name": "Chilli con Carne",
            "updated_at": "2019-07-29T22:49:47.519Z",
          },
        ]
      }
      assemblyMap={
        Object {
          "1": Object {
            "created_at": "2019-07-29T22:49:47.431Z",
            "id": 1,
            "name": "Souvlaki",
            "updated_at": "2019-07-29T22:49:47.431Z",
          },
          "2": Object {
            "created_at": "2019-07-29T22:49:47.519Z",
            "id": 2,
            "name": "Chilli con Carne",
            "updated_at": "2019-07-29T22:49:47.519Z",
          },
        }
      }
      concern="ITEM"
      csrf="3yIq75tyV791TFkzSZo/S4b+TVa/N5J3dI14PzMUPviGViEGeDSnGnvNix+MaB0RB0XuV7bpX8I3SmIEAr4/zQ=="
      departmentMap={
        Object {
          "3": Object {
            "created_at": "2019-07-21T16:03:32.883Z",
            "id": 3,
            "name": "Bakery",
            "updated_at": "2019-07-21T16:03:32.883Z",
          },
          "5": Object {
            "created_at": "2019-07-21T16:03:32.883Z",
            "id": 5,
            "name": "Meat",
            "updated_at": "2019-07-21T16:03:32.883Z",
          },
        }
      }
      departments={
        Array [
          Object {
            "created_at": "2019-07-21T16:03:32.883Z",
            "id": 3,
            "name": "Bakery",
            "updated_at": "2019-07-21T16:03:32.883Z",
          },
          Object {
            "created_at": "2019-07-21T16:03:32.883Z",
            "id": 5,
            "name": "Meat",
            "updated_at": "2019-07-21T16:03:32.883Z",
          },
        ]
      }
      itemMap={
        Object {
          "13": Object {
            "created_at": "2019-07-21T16:03:33.162Z",
            "department_id": 3,
            "id": 13,
            "name": "Agave Syrup",
            "updated_at": "2019-07-21T16:03:33.162Z",
          },
          "15": Object {
            "created_at": "2019-07-21T16:03:33.162Z",
            "department_id": 5,
            "id": 15,
            "name": "Agave Syrup",
            "updated_at": "2019-07-21T16:03:33.162Z",
          },
        }
      }
      orderItems={
        Array [
          Object {
            "assembly_id": 1,
            "created_at": "2019-07-29T22:50:37.840Z",
            "department_id": 3,
            "id": 22,
            "item_id": 15,
            "order_id": 11,
            "picked": false,
            "qty": 3,
            "updated_at": "2019-07-29T22:50:37.840Z",
          },
          Object {
            "assembly_id": null,
            "created_at": "2019-07-29T22:50:37.887Z",
            "department_id": 5,
            "id": 23,
            "item_id": 13,
            "order_id": 11,
            "picked": false,
            "qty": 1,
            "updated_at": "2019-07-29T22:50:37.887Z",
          },
        ]
      }
    />
  </div>
</Fragment>
`;
